stdlib/Logging/docs/src/index.md

```diff
@@ -0,0 +1,202 @@
+# Logging
+
+The `Logging` module provides a way to record the history and progress of a
+computation as a log of events.  Events are created by inserting a logging
+statement into the source code, for example:
+
+```julia
+@warn "Abandon printf debugging, all ye who enter here!"
+┌ Warning: Abandon printf debugging, all ye who enter here!
+└ @ Main REPL[1]:1
+```
+
+The system provides several advantages over peppering your source code with
+calls to `println()`.  First, it allows you to control the visibility and
+presentation of messages without editing the source code.  For example, in
+contrast to the `@warn` above
+
+```julia
+@debug "The sum of some values $(sum(rand(100)))"
+```
+
+will produce no output by default.  Furthermore, it's very cheap to leave debug
+statements like this in the source code because the system avoids evaluating
+the message if it would later be ignored.  In this case `sum(rand(100))` and
+the associated string processing will never be executed unless debug logging is
+enabled.
+
+Second, the logging tools allow you to attach arbitrary data to each event as a
+set of key value pairs. This allows you to capture local variables and other
+program state for later analysis. For example, to attach the local array
+variable `A` and the sum of a vector `v` as the key `s` you can use
+
+```jldoctest
+A = ones(Int, 4, 4)
+v = ones(100)
+@info "Some variables"  A  s=sum(v)
+
+# Output
+┌ Info: Some variables
+│   A =
+│    4×4 Array{Int64,2}:
+│     1  1  1  1
+│     1  1  1  1
+│     1  1  1  1
+│     1  1  1  1
+└   s = 100.0
+```
+
+All of the logging macros `@debug`, `@info`, `@warn` and `@error` share common
+features which are described in detail in the documentation for the more
+general macro [`@logmsg`](@ref).
+
+## Log event structure
+
+Each event generates several pieces of data, some provided by the user and some
+automatically extracted. Let's examine the user-defined data first:
+
+* The *log level* is a broad category for the message which is used for early
+  filtering. There are several standard levels of type [`LogLevel`](@ref);
+  user defined levels are also possible.
+  - Use `Debug` for verbose information which could be useful when debugging an
```

Comment:
I'd use "that" instead of "which" here.

Reply:
Aha, I've already removed most of these "which"s locally.  Will push an update shortly.

Reply:
Excellent, you've clearly got your pronouns under control then ;)